File: cpp/1604. 警告一小时内使用相同员工卡大于等于三次的人.cpp
class Solution {
public:
    vector<string> alertNames(vector<string>& keyName, vector<string>& keyTime) {
        unordered_map<string,vector<int> > timeMap;
        for(int i=0;i<keyName.size();i++){
            string name=keyName[i];
            string time=keyTime[i];
            int hour=(time[0]-'0')*10+(time[1]-'0');
            int minute=(time[3]-'0')*10+(time[4]-'0');
            timeMap[name].emplace_back(hour*60+minute);
        }
        vector<string> ans;
        for(auto &[name,vec]:timeMap){
            sort(vec.begin(),vec.end());
            int size=vec.size();
            for(int i=2;i<size;i++){
                int time1=vec[i-2],time2=vec[i];
                int diff=time2-time1;
                if(diff<=60){
                    ans.emplace_back(name);
                    break;
                }
            }
        }
        sort(ans.begin(),ans.end());
        return ans;
    }
};

File: 刷题/golang/1604. 警告一小时内使用相同员工卡大于等于三次的人.go
func alertNames(keyName []string, keyTime []string) (ans []string) {
    timeMap:=make(map[string][]int)
    for i,name:=range keyName{
        time:=keyTime[i]
        hour:=int(time[0]-'0')*10+int(time[1]-'0')
        minute:=int(time[3]-'0')*10+int(time[4]-'0')
        timeMap[name]=append(timeMap[name],hour*60+minute)
    }
    for name,vec:=range timeMap{
        sort.Ints(vec)
        for i,t:=range vec[2:]{
            diff:=t-vec[i]
            if diff<=60{
                ans=append(ans,name)
                break
            }
        }
    }
    sort.Strings(ans)
    return
}